src/objects/Arrayable.ts

```diff
@@ -147,15 +147,17 @@ class Arrayable<T> extends Array<T> {
         return this.constructor.from(clonedeep([...this])) as Arrayable<T>
     }
 
-    groupBy(key: string | Function) {
-        const grouped = this.reduce<{ [key: string]: Arrayable<T> }>((result, item) => {
+    groupBy<K>(key: string | Function) {
```

Comment:
While we are at it, we can also type the arguments.

```suggestion
    groupBy<K extends keyof T>(key: K | ((item: T) => T[K]) ) {
```

Then replace `Mappable<K>` with `Mappable<T[K]>`.

I think this would work, what are your thoughts?